src/main/java/tool/Command.java

```diff
@@ -0,0 +1,161 @@
+package tool;
+
+import command.Deadlines;
+import command.Events;
+import command.Task;
+import command.ToDos;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+public class Command {
+    private String command;
+    private String arguments;
+
+    public Command(String command){
+        this.command = command;
+    }
+
+    public Command(String command, String arguments){
+        this.command = command;
+        this.arguments = arguments;
+    }
+
+    public boolean execute(TaskList taskList, UI ui) {
+        switch (command) {
+            case "bye":
+                ui.print("Bye. Hope to see you again soon!");
+                return true;
+            case "list":
+                ui.print("Here are the tasks in your list:");
+                for (int i = 0; i < taskList.size(); i++) {
+                    ui.print(String.valueOf(i + 1) + "." + taskList.get(i));
+                }
+                return false;
+            case "find":
+                ui.print("Here are the matching tasks in your list:");
+                int count = 0;
+                for (int i = 0; i < taskList.size(); i++) {
+                    if (taskList.get(i).containsString(arguments)){
+                        count++;
+                        ui.print(String.valueOf(count) + "." + taskList.get(i));
+                    }
+                }
+                return false;
+            case "done":
+                try {
+                    int num = Integer.parseInt(arguments);
+                    if (taskList.size() == 0) {
+                        ui.printError("☹ OOPS!!! You have no tasks currently.");
+                    } else if (num <= 0) {
+                        ui.printError("☹ OOPS!!! Task number cannot be equal to or less than 0");
+                    } else if (num > taskList.size()) {
+                        ui.printError("☹ OOPS!!! The number of a done command cannot be greater than the number of tasks " +
+                                "you have.");
+                    } else {
+                        taskList.get(num - 1).markAsDone();
+                        ui.print("Nice! I've marked this task as done:");
+                        ui.print(taskList.get(num - 1).toString());
+                    }
+                } catch (NumberFormatException ex) {
+                    //handle exception here
```

Comment:
With better handling for error cases, you can improve the usability of your program.